cannon/mipsevm/state.go

```diff
@@ -70,6 +72,214 @@ func (s *State) EncodeWitness() StateWitness {
 	return out
 }
 
+func (s *State) Serialize(out io.Writer) error {
```

Comment:
The state is so small (< 500 bytes) that it may just be worth serializing/deserializing the state witness, and writing/reading it as one blob into the writer, to avoid many different write/read calls.